shapely/tests/test_strtree.py

```diff
@@ -65,151 +65,231 @@ def poly_tree():
+        [Point(0, 0), "still not a geometry"],
+        [[], "in a mixed array", 1],
+    ],
+)
+@pytest.mark.filterwarnings("ignore:Creating an ndarray from ragged nested sequences:")
+def test_query_invalid_geometry(tree, geometry):
     with pytest.raises(TypeError):
-        tree.query_items("I am not a geometry")
-
+        tree.query(geometry)
 
-def test_query_none(tree):
-    assert tree.query_items(None).size == 0
 
-
-@pytest.mark.parametrize("geometry", [empty, empty_point, empty_line_string])
-def test_query_empty(tree, geometry):
-    assert tree.query_items(geometry).size == 0
+def test_query_invalid_dimension(tree):
+    with pytest.raises(TypeError, match="Array should be one dimensional"):
+        tree.query([[Point(0.5, 0.5)]])
 
 
 @pytest.mark.parametrize(
     "tree_geometry, geometry,expected",
     [
-        ([point], box(0, 0, 10, 10), [0]),
-        # None is ignored in the tree, but the index of the valid geometry should
-        # be retained.
+        # Empty tree returns no results
+        ([], point, []),
+        ([], [point], [[], []]),
+        ([], None, []),
+        ([], [None], [[], []]),
+        # Tree with only None returns no results
+        ([None], point, []),
+        ([None], [point], [[], []]),
+        ([None], None, []),
+        ([None], [None], [[], []]),
+        # querying with None returns no results
+        ([point], None, []),
+        ([point], [None], [[], []]),
+        # Empty is included in the tree, but ignored when querying the tree
+        ([empty], empty, []),
+        ([empty], [empty], [[], []]),
+        ([empty], point, []),
+        ([empty], [point], [[], []]),
+        ([point, empty], empty, []),
+        ([point, empty], [empty], [[], []]),
+        # None and empty are ignored in the tree, but the index of the valid
+        # geometry should be retained.
         ([None, point], box(0, 0, 10, 10), [1]),
+        ([None, point], [box(0, 0, 10, 10)], [[0], [1]]),
         ([None, empty, point], box(0, 0, 10, 10), [2]),
+        ([point, None, point], box(0, 0, 10, 10), [0, 2]),
+        ([point, None, point], [box(0, 0, 10, 10)], [[0, 0], [0, 2]]),
+        # Only the non-empty query geometry gets hits
+        ([empty, point], [empty, point], [[1], [1]]),
+        (
+            [empty, empty_point, empty_line_string, point],
+            [empty, empty_point, empty_line_string, point],
+            [[3], [3]],
+        ),
     ],
 )
-def test_query(tree_geometry, geometry, expected):
-    tree = shapely.STRtree(np.array(tree_geometry))
-    assert_array_equal(tree.query_items(geometry), expected)
+def test_query_with_none_and_empty(tree_geometry, geometry, expected):
+    tree = STRtree(tree_geometry)
+    assert_array_equal(tree.query(geometry), expected)
 
 
 @pytest.mark.parametrize(
     "geometry,expected",
     [
         # points do not intersect
-        (shapely.points(0.5, 0.5), []),
+        (Point(0.5, 0.5), []),
+        ([Point(0.5, 0.5)], [[], []]),
```

Comment:
Pretty much every test case now has both a scalar and array variant.  In the case of no predicate, `interesects`, `dwithin`, these were migrated in from the associated `query_bulk` tests; in the other predicates they were added in this PR since they didn't have query_bulk specific tests.